nvdaHelper/remote/log.cpp

```diff
@@ -12,13 +12,56 @@ This license can be found at:
 http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 */
 
+#include <string>
+#include <queue>
 #include <crtdbg.h>
 #include "nvdaControllerInternal.h"
+#include "nvdaHelperRemote.h"
 #include <common/log.h>
 
+std::deque<std::tuple<int, std::wstring>> logQueue;
+
+// Fetch a  maximum number of messages from the log queue
+// and send them onto NvDA via rpc.
+void __stdcall log_flushQueue(ULONG_PTR maxCount) {
+	while(!logQueue.empty()) {
+		if(maxCount < 1) break;
+		{

```

Comment:
A for loop can have multiple conditions, and (I think) it communicates intent more clearly, since the conditions and logic for the extent of the loop are all in one place:

```suggestion
	for(;maxCount > 0 && !logQueue.empty(); --maxCount) {
		{  // scope limit for reference items returned by front()
```

As a side note, `break` on the same line as `if`, with a brace on the following line is really easy to miss-read. The brace looks like the body of the `if`.